### ISSUE TITLE
CUDA Memory cannot release on "cv/example_demo.ipynb"

### ISSUE THREAD

**Issue by chufanwu** [lichengunc/MAttNet#27]
Hi~
I've trying to migrate the code at  `cv/example_demo.ipynb` to form a detection network. However, I found that GPU memory cannot be released when running`mattnet.forward_image(img_dir)` several times to detect multiple images. I traced the code carefully and added lots of `del` and `torch.cuda.empty_cache() ` but it's still not working. I even tried to delete mattnet instance and create a new one each time for detection but it's still not working. Obviously some variables are not released. Since you are using PyTorch 0.2, I am using 0.4.0, I thought it would be a GPU Variable release problem causing by version difference. Did 0.2 have the same problem? Thanks.

Best,
Chufan

**Comment by lichengunc** [comment]
Hi Chufan, it works for v0.2 (and the demo). I saw quite a few incompatible issues here, perhaps the simplest way is to switch to using old version :-)

**Comment by chufanwu** [comment]
Hi @lichengunc , Thanks for your reply. I have downgraded to `PyTorch 0.2.0` with `CUDA 8.0` and `cudnn 6.0.21`. I tried to run `cv/mrcn_dection` but got these on test_predict1 part.
```
 UserWarning: PyTorch was compiled without cuDNN support. To use cuDNN, rebuild PyTorch making sure the library is visible to the build system.
  "PyTorch was compiled without cuDNN support. To use cuDNN, rebuild 
```
I installed PyTorch through conda with `conda install pytorch=0.2 cuda80 -c pytorch`. Are you using cuda75? It seems your code cannot detect the installed cudnn. Thanks!
